### PR TITLE
Setup tools for the C++ sim

### PR DESCRIPTION
**Pull request by kmkar31**
This PR address the following:
- creates a package for the C++ sim

### CODE REVIEW

**Comment by kmkar31** [comment]
Main changes to note:
- Popen calls in run_job.py and run_trials.py

argusim was not being recognised because the Popen call did not pass the environment to later scripts. This was fixed by creating a copy of os.environ in run_job.py and passing it forward to run_trials and sim.py

File: argusim/run_job.py [hunk 1]
```diff
@@ -60,21 +54,6 @@
...
-        ],
-        cwd=os.path.dirname(pose_simulator_build_script_abs),
-    )
-    while process.poll() is None:  # wait while the code is being built
-        time.sleep(0.1)
-    return_code = process.returncode
-    if return_code != 0:
-        raise AssertionError(RED + f"Build failed with code {return_code}" + RESET)
-
-
```
Comment (2shaar2059) [inline argusim/run_job.py:77]: why aren't we building the simulator anymore?

Comment (kmkar31) [reply argusim/run_job.py:77]: It isn't great to couple building and running the sim. I can write another 'debug.sh' that first builds and then runs if that'll help. The point is though, I want to keep building away from run_job

Comment (kmkar31) [reply argusim/run_job.py:77]: Added a run_debug.sh script that build sim before running monte carlo

**Review by prchall123 — CHANGES_REQUESTED** [review]
to run I had to clear the cache everywhere before building. Plotting fails with a free(): invalid pointer. Debugging from the run_debug.sh doesn't allow for breakpointing. Running run_jobs.py separately with debugpy isn't trivial to set up. Could you please make a launch.json entry for doing so? We should understand why the plotting function is causing this problem and if there's a way to fix it.

**Review by prchall123 — APPROVED** [review]
worked, thanks!

**Pull request merged**